test/sth_app_test.js

```diff
@@ -1,3 +1,222 @@
+  (sthTestConfig, sthConfig, sthApp.sthDatabase, sthHelper);
+  var hapi = require('hapi');
+  var request = require('request');
+  var expect = require('expect.js');
+
+  console.log('*** Running the server tests with the following environment variables:');
+  console.log('***   - SAMPLES: %s', sthTestConfig.SAMPLES);
+  console.log('***   - HOST: %s', sthConfig.HOST);
+  console.log('***   - PORT: %s', sthConfig.PORT);
+  console.log('***   - DB_USERNAME: %s', sthConfig.DB_USERNAME);
+  console.log('***   - DB_PASSWORD: %s', sthConfig.DB_PASSWORD);
+  console.log('***   - DB_HOST: %s', sthConfig.DB_HOST);
+  console.log('***   - DB_PORT: %s', sthConfig.DB_PORT);
+  console.log('***   - DB_NAME: %s', sthConfig.DB_NAME);
+  console.log('***   - ENTITY_ID: %s', sthTestConfig.ENTITY_ID);
+  console.log('***   - ATTRIBUTE_ID: %s', sthTestConfig.ATTRIBUTE_ID);
+  console.log('***   - TYPE: %s', sthTestConfig.TYPE);
+  console.log('***   - START_DATE: %s', sthTestConfig.START_DATE);
+  console.log('***   - END_DATE: %s', sthTestConfig.END_DATE);
+  console.log('***   - MIN_VALUE: %s', sthTestConfig.MIN_VALUE);
+  console.log('***   - MAX_VALUE: %s', sthTestConfig.MAX_VALUE);
+
+  describe('database connection', function() {
+    it('should be a database available', function(done) {
+      sthApp.sthDatabase.connect(sthConfig.DB_AUTHENTICATION, sthConfig.DB_HOST,
+        sthConfig.DB_PORT, sthConfig.DB_NAME, function(err) {
+          done(err);
+        });
+
+    });
+  });
+
+  describe('database clean up', function() {
+    it('should drop the event raw data collection if it exists',
+      sthTestHelper.dropRawEventCollectionTest);
+
+    it('should drop the aggregated data collection if it exists',
+      sthTestHelper.dropAggregatedDataCollectionTest);
+  });
+
+  describe('server start', function() {
+    it('should start gracefully', function(done) {
+      sthApp.sthServer.startServer(
+        sthConfig.HOST,
+        sthConfig.PORT,
+        sthApp.sthDatabase,
+        function(err, server) {
+          expect(err).to.equal(null);
+          expect(server).to.be.a(hapi.Server);
+          done();
+        });
+    });
+  });
+
+  describe('invalid routes', function() {
+    it('should respond with 404 - Not Found if invalid HTTP method', function(done) {
+      request({
+        uri: sthTestHelper.getValidURL(),
+        method: 'POST'
+      }, function(err, response, body) {
+        var bodyJSON = JSON.parse(body);
+        expect(err).to.equal(null);
+        expect(response.statusCode).to.equal(404);
+        expect(bodyJSON.statusCode).to.equal(404);
+        expect(response.statusMessage).to.equal('Not Found');
+        expect(bodyJSON.error).to.equal('Not Found');
+        done();
+      });
+    });
+
+    it('should respond with 404 - Not Found if invalid path', function(done) {
+      request({
+        uri: sthTestHelper.getInvalidURL({
+          invalidPath: true
+        }),
+        method: 'GET'
```

Comment:
Yeah, that was one of my questions in our last meeting with @fgalan and @dmoranj regarding using the NGSI format for the requests. If I am not wrong, we decided to go for the GET (convenience) version instead of for the POST version to avoid defining a way to pass the query params the STH API takes in the POST payload (in case that would even be possible according to the spec :) ).